### PR TITLE
[P8-62] Add robot grid lcgs program

### PR DESCRIPTION
**Pull request by NicEastvillage**
In this scenario, 4 robots exists on an NxN grid, starting on separate corners.
```
+----+----+----+----+
| r3 |    |    | r2 |
+----+----+----+----+
|    |    |    |    |
+----+----+----+----+
|    |    |    |    |
+----+----+----+----+
| r1 |    |    | r4 |
+----+----+----+----+
```
Each robot wants to move to the opposite corner of where they start, every step either standing still or moving up, down, left, or right. However, if two robots ever share the same position or move through each other, they crash into each other and can no longer move.

Included ATL queries are:
* Can r1 and r2 swap places with help from r3 (FALSE expected): `<<r1,r2,r3>> F (r1.at_home & !r1.crashed & r2.at_home & !r2.crashed)`
* Can r1 get past r2 when helped by r3 and r4 (TRUE expected): `<<r1,r3,r4>> F (r1.at_home & !r1.crashed)`
* Can everyone get to their target if they word together (TRUE expected): `<<r1,r2,r3,r4>> F (!any_crashes & r1.at_target & r2.at_target & r3.at_target & r4.at_target)`

### CODE REVIEW

**Review by GeneralDenmark — APPROVED** [review]
Super fedt med flere versioner, og det ser også ud til at spillepladens størrelse er relativ at denne er nem at skalere på, ved at ændre i værdien for N.

File: docs/lcgs/working-examples/Robot_Grid/Robot_Grid_N3.lcgs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        || (y - 1 == OTHER_2.y && move_down  && OTHER_2.move_up)
+        // Does this and OTHER_3 move through each other?
+        || (x + 1 == OTHER_3.x && move_right && OTHER_3.move_left)
+        || (x - 1 == OTHER_3.x && move_left  && OTHER_3.move_right)
+        || (y + 1 == OTHER_3.y && move_up    && OTHER_3.move_down)
+        || (y - 1 == OTHER_3.y && move_down  && OTHER_3.move_up)
+    ;
+
+    label at_start = x == START_X && y == START_Y;
+    label at_target = x == TARGET_X && y == TARGET_Y;
```
Comment (lars4509) [inline docs/lcgs/working-examples/Robot_Grid/Robot_Grid_N3.lcgs:65]: The formulas use the proposition `at_home`, which is not to be found in this program.

File: docs/lcgs/working-examples/Robot_Grid/Robot_Grid_N4.lcgs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        || (y - 1 == OTHER_2.y && move_down  && OTHER_2.move_up)
+        // Does this and OTHER_3 move through each other?
+        || (x + 1 == OTHER_3.x && move_right && OTHER_3.move_left)
+        || (x - 1 == OTHER_3.x && move_left  && OTHER_3.move_right)
+        || (y + 1 == OTHER_3.y && move_up    && OTHER_3.move_down)
+        || (y - 1 == OTHER_3.y && move_down  && OTHER_3.move_up)
+    ;
+
+    label at_start = x == START_X && y == START_Y;
+    label at_target = x == TARGET_X && y == TARGET_Y;
```
Comment (lars4509) [inline docs/lcgs/working-examples/Robot_Grid/Robot_Grid_N4.lcgs:61]: Just as mentioned before the `at_home` proposition is not described in the program.

**Review by lars4509 — APPROVED** [review]
Cool, concise program. Have one comment about propositions, otherwise it looks good to me.

File: docs/lcgs/working-examples/Robot_Grid/Robot_Grid_N3.lcgs (hunk 1 above)
Comment (NicEastvillage) [reply docs/lcgs/working-examples/Robot_Grid/Robot_Grid_N3.lcgs]: Whoops. Well spotted. I renamed it at some point. Fixed in e848b95

**Pull request merged**